Add OLM 1.0 relnote for 4.14 (#63605)
openshift/openshift-docs: release_notes/ocp-4-14-release-notes.adoc

[id="ocp-4-14-olm"]
=== Operator lifecycle

[id="ocp-4-14-olmv1"]
==== Operator Lifecycle Manager v1 (Technology Preview)

{product-title} 4.14 introduces components for a next-generation iteration of Operator Lifecycle Manager (OLM) as a Technology Preview feature. Known during this phase as OLM v1, the updated framework evolves many of the concepts that have been part of the version of OLM included in OpenShift Container Platform up to this point, referred to retroactively as OLM v0.

OLM v1 improves Operator and extension lifecycle management in the following areas:

* Dependencies and constraints
* Tenant isolation
* Simplified packaging models
* Multi-cluster support

In OLM v1, a _plain bundle_ is a static collection of arbitrary Kubernetes manifests in the YAML format. In the Technology Preview phase, cluster administrators and Operator catalog maintainers can build and publish a file-based catalog that includes a plain bundle image by completing the following procedures:

. Build a plain bundle image.
. Create a file-based catalog.
. Add the plain bundle image to your file-based catalog.
. Build your catalog as an image.
. Publish your catalog image.

> For more information, see "Managing catalogs for OLM v1".

Author: `xref` to content to be added after https://github.com/openshift/openshift-docs/pull/63086 merges.

[id="ocp-4-14-olm"]
=== Operator lifecycle

[id="ocp-4-14-olmv1"]
==== Operator Lifecycle Manager v1 (Technology Preview)

> {product-title} 4.14 introduces components for a next-generation iteration of Operator Lifecycle Manager (OLM) as a Technology Preview feature. Known during this phase as OLM v1, the updated framework evolves many of the concepts that have been part of the version of OLM included in OpenShift Container Platform up to this point, referred to retroactively as OLM v0.

Reviewer: Nit:

```suggestion
{product-title} 4.14 introduces components for a next-generation iteration of Operator Lifecycle Manager (OLM) as a Technology Preview feature. Known during this phase as OLM v1, the updated framework evolves many of the concepts that have been part of the version of OLM included in {product-title} up to this point, referred to retroactively as OLM v0.
```

[id="ocp-4-14-olm"]
=== Operator lifecycle

[id="ocp-4-14-olmv1"]
==== {olmv1-first} (Technology Preview)

Operator Lifecycle Manager (OLM) has been included with {product-title} 4 since its initial release. {product-title} {product-version} introduces components for a next-generation iteration of OLM as a Technology Preview feature, known during this phase as _{olmv1}_. This updated framework evolves many of the concepts that have been part of previous versions of OLM and adds new capabilities.

During this Technology Preview phase of {olmv1} in {product-title} 4.14, administrators can explore the following features:

Fully declarative model that supports GitOps workflows::
{olmv1} simplifies Operator management through two key APIs:
+
--

> * The `Operator` API, provided by the new Operator Controller component, streamlines management of installed Operators by consolidating user-facing APIs into a single object. This empowers administrators and SREs to automate processes and define desired states by using GitOps principles.

Reviewer 2: Actually, OLM has the `Operator` kind before, which is from the CRD `operators.operators.coreos.com`. Here is the new `Operator` kind from CRD: `operators.operators.operatorframework.io`. Both of them use the `operator` strings for now.
See the discussion on Slack: https://redhat-internal.slack.com/archives/GHMALGJV6/p1689573773633059 or the https://issues.redhat.com/browse/OCPBUGS-22094
```console
MacBook-Pro:~ jianzhang$ oc get operators.operators.operatorframework.io 
NAME           AGE
quay-example   4h10m
MacBook-Pro:~ jianzhang$ oc get operators.operators.coreos.com 
NAME                                                AGE
cluster-logging.openshift-logging                   7h13m
elasticsearch-operator.openshift-operators-redhat   7h13m
loki-operator.openshift-operators-redhat            7h13m
oadp-operator.e2e-test-olm-a-fwobi02w-gsc8h         3h42m
```
So, to avoid confusion, here I'd suggest using the `operators.operators.operatorframework.io` instead of `operator`.

Author: Good point! I had created this [olmv1-operator-api-group.adoc](https://github.com/openshift/openshift-docs/pull/65963/files#diff-a81ceaf5428923a809911af7ed6b606a17e60003ee16e3926cf87c64384ddc55) `[NOTE]` snippet in another WIP PR, which talks about that difference, so I'll re-use that here as well.

* The `Catalog` API, provided by the new Catalogd component, serves as the foundation for {olmv1}, unpacking catalogs for on-cluster clients so that users can discover installable content, such as Operators and Kubernetes extensions. This provides increased visibility into all available Operator bundle versions, including their details, channels, and update edges.
--
+

> For more information, see _Operator Controller_ and _Catalogd_.

Reviewer 3: Did you mean to add xrefs here?

Author: Sorry, yea, mentioned above in https://github.com/openshift/openshift-docs/pull/63605#issuecomment-1769245183.

Improved control over Operator updates::

> With the greater insight into catalog content, administrators can specify target versions for installation and updates. This grants administrators more control over the target version of Operator updates. For more information, see _Installing an Operator from a catalog_.

Reviewer 3: Just a wording suggestion:
```suggestion
With improved insight into catalog content, administrators can specify target versions for installation and updates. This grants administrators more control over the target version of Operator updates. For more information, see _Installing an Operator from a catalog_.
```
also see link question above

Flexible Operator packaging format::
Administrators can use file-based catalogs to install and manage the following types of content:
+
--
* OLM-based Operators, similar to the existing OLM experience
* _Plain bundles_, which are static collections of arbitrary Kubernetes manifests
--
+

> In addition, bundle size is no longer constrained by the etcd value size limit. For more information, see _Managing catalogs in {olmv1}_.

Reviewer 3: same q about links

[NOTE]
====

> For {product-title} 4.14, documented procedures for {olmv1} are CLI-based only. Alternatively, administrators can create and view related objects in the web console by using normal methods, such as the *Import YAML* and *Search* pages. However, the existing *OperatorHub* and *Installed Operators* pages do not yet observe {olmv1} components.

Reviewer 3: ```suggestion
For {product-title} {product-version}, documented procedures for {olmv1} are CLI-based only. Alternatively, administrators can create and view related objects in the web console by using normal methods, such as the *Import YAML* and *Search* pages. However, the existing *OperatorHub* and *Installed Operators* pages do not yet observe {olmv1} components.
```

====

> For more information, see _{olmv1-first}_.

Reviewer 3: link q

|====

[discrete]
=== Operator Technology Preview features
=== Operator lifecycle and development Technology Preview features

.Operator Technology Preview tracker
.Operator lifecycle and development Technology Preview tracker
[cols="4,1,1,1",options="header"]
|====
|Feature |4.12 |4.13 |4.14

|Hybrid Helm Operator
|Technology Preview
|Technology Preview
|Operator Lifecycle Manager (OLM) v1
|Not Available
|Not Available
|Technology Preview

|Java-based Operator
|RukPak
|Technology Preview
|Technology Preview
|Technology Preview

> |Multi-cluster Engine Operator

Author: I am not sure why "Multi-cluster Engine Operator" is here in the "Operator lifecycle and development Technology Preview tracker", so I moved it down to the "Architecture Technology Preview features" section since it seemed like it is related to hosted control planes, but I will follow up with the related tech writer to confirm that before merge.

|Technology Preview
|Technology Preview
|Platform Operators
|Technology Preview

> |Node Observability Operator

Author: Removing "Node Observability Operator" from the "Operator lifecycle and development Technology Preview tracker" because it is a duplicate here and is already represented (and is a better fit) over in the "Scalability and performance Technology Preview features" section.

|====

[discrete]
[id="ocp-4-14-operators-tech-preview"]
=== Operator lifecycle and development Technology Preview features

.Operator lifecycle and development Technology Preview tracker
[cols="4,1,1,1",options="header"]
|====
|Feature |4.12 |4.13 |4.14

|Operator Lifecycle Manager (OLM) v1
|Not Available
|Not Available
|Technology Preview

|RukPak
|Technology Preview
|Technology Preview
|Technology Preview

|Platform Operators
|Technology Preview
|Technology Preview
|Technology Preview

> |Hybrid Helm Operator

Reviewer 2: I guess the `Hybrid Helm Operator` belongs to the `Operator-SDK`, not OLM.

|Technology Preview
|Technology Preview
|Technology Preview

> |Java-based Operator

Reviewer 2: Actually, the `Java-based Operator` belongs to the `operator-sdk`, not OLM.
